### PR TITLE
PLU-312: feat: support telegram topics

### PR DESCRIPTION
**Pull request by m0nggh**
## Problem

Users cannot send message to telegram topics in their groups.

## Details
- Add `message_thread_id` to allow users to do so.
- Add error catching for invalid topic id and unit test

TODOs:
- [x] Add to user guide on how to obtain topic id

## Tests
- [x] Sending message to telegram group still works
- [x] Sending message to valid topic works
- [x] Sending message to invalid topic throws step error

### CODE REVIEW

File: packages/backend/src/apps/telegram-bot/actions/send-message/index.ts [hunk 1]
```diff
@@ -47,6 +47,15 @@ const action: IRawAction = {
       description:
         'Sends the message silently. Users will receive a notification with no sound.',
     },
+    {
+      label: 'Topic ID',
+      key: 'topicId',
+      type: 'string' as const,
```
Comment (ogp-weeloong) [inline packages/backend/src/apps/telegram-bot/actions/send-message/index.ts:53]: Can we time box 1 hour to convert this to allowArbitrary dropdown, as discussed.

Thanks!

**Review by ogp-weeloong — APPROVED** [review]
I approve first, try to do dropwn if you can otherwise nvm.

File: packages/backend/src/apps/telegram-bot/actions/send-message/index.ts [hunk 2]
```diff
@@ -47,6 +47,26 @@ const action: IRawAction = {
       description:
         'Sends the message silently. Users will receive a notification with no sound.',
     },
+    {
+      label: 'Topic ID',
```
Comment (ogp-weeloong) [inline packages/backend/src/apps/telegram-bot/actions/send-message/index.ts]: nit: maybe `Channel Topic`?

File: packages/backend/src/apps/telegram-bot/actions/send-message/index.ts [hunk 3]
```diff
@@ -47,6 +47,26 @@ const action: IRawAction = {
...
+      key: 'topicId',
+      type: 'dropdown' as const,
+      allowArbitrary: true,
+      required: false,
+      description:
```
Comment (ogp-weeloong) [inline packages/backend/src/apps/telegram-bot/actions/send-message/index.ts:55]: nit: maybe can leave out?

File: packages/backend/src/apps/telegram-bot/dynamic-data/get-topic-ids.ts [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+  return { name: forumTopic.name, id: messageObject.message_thread_id }
+}
+
+const dynamicData: IDynamicData = {
+  key: 'getTelegramTopicIds',
+  name: 'Get Telegram Topic IDs',
+  async run($: IGlobalVariable): Promise<DynamicDataOutput> {
+    const topicIdsMap: { name: string; value: string }[] = []
+    const topicIdsSet = new Set<number>()
```
Comment (ogp-weeloong) [inline packages/backend/src/apps/telegram-bot/dynamic-data/get-topic-ids.ts:44]: nit: actually can just re-use map

File: packages/backend/src/apps/telegram-bot/dynamic-data/get-topic-ids.ts [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    messageObject?.forum_topic_created || messageObject?.forum_topic_edited
+
+  return { name: forumTopic.name, id: messageObject.message_thread_id }
+}
+
+const dynamicData: IDynamicData = {
+  key: 'getTelegramTopicIds',
+  name: 'Get Telegram Topic IDs',
+  async run($: IGlobalVariable): Promise<DynamicDataOutput> {
+    const topicIdsMap: { name: string; value: string }[] = []
```
Comment (ogp-weeloong) [inline packages/backend/src/apps/telegram-bot/dynamic-data/get-topic-ids.ts]: nit: not a map

**Review by ogp-weeloong — COMMENTED** [review]
As discussed - lets put it back to static text input field, because it turns out the bot API can't handle topic renames or deletes. So the dynamic drop down will be too confusing for users.

Thanks for putting in the effort to change to dynamic drop down, my bad on suggesting this.

File: packages/backend/src/apps/telegram-bot/actions/send-message/index.ts (hunk 2 above)
Comment (m0nggh) [reply packages/backend/src/apps/telegram-bot/actions/send-message/index.ts]: ok thanks

**Pull request merged**